tools/osx-build-cmake.sh

```diff
@@ -0,0 +1,230 @@
+    -DWITH_WIN32UI=OFF
+    -DWITH_QT=OFF
+    -DWITH_GTK=OFF
+    # -DWITH_MSMF=OFF
+  )
+fi
+
+# Clear ccache stats
+ccache -z
+
+# Measure build time
+cmake "${CMAKE_OPTS[@]}" ..
+# $ pkg-config --libs opencv4 | sed -e 's| |\n|g' | sed -e 's|^-l||g' | tac
+CV_MODULES=(
+  opencv_core
+  opencv_imgproc
+  opencv_photo
+  opencv_xphoto
+  opencv_flann
+  opencv_features2d
+  opencv_imgcodecs
+  opencv_calib3d
+  opencv_objdetect
+  opencv_xobjdetect
+  opencv_video
+  opencv_ximgproc
+  opencv_ml
+  opencv_shape
+  opencv_xfeatures2d
+  opencv_viz
+  opencv_videoio
+  opencv_videostab
+  opencv_plot
+  opencv_dnn
+  opencv_text
+  opencv_datasets
+  opencv_tracking
+  opencv_surface_matching
+  opencv_optflow
+  opencv_superres
+  opencv_phase_unwrapping
+  opencv_structured_light
+  opencv_stereo
+  opencv_saliency
+  opencv_rgbd
+  opencv_reg
+  opencv_rapid
+  opencv_quality
+  opencv_mcc
+  opencv_line_descriptor
+  opencv_intensity_transform
+  opencv_img_hash
+  opencv_hfs
+  opencv_hdf
+  opencv_fuzzy
+  opencv_freetype
+  opencv_face
+  opencv_highgui
+  opencv_dpm
+  opencv_dnn_superres
+  opencv_dnn_objdetect
+  opencv_cvv
+  opencv_ccalib
+  opencv_bioinspired
+  opencv_bgsegm
+  opencv_aruco
+  opencv_alphamat
+  opencv_stitching
+  opencv_gapi
+)
+for m in "${CV_MODULES[@]}"; do
+    if make help | grep "$m"; then
+        time make -j "$m"
+    fi
+done
+time make -j
+
+# Print ccache stats
+ccache -s
+
+# Copy compiled python module and 3rd-party stuff to package
+cp -f lib/python3/cv2*.so "$REPO/$NAME/cv2"
+
+# Copy dependencies
+dep_libs=()
+
+# There are libs in /System, /usr/lib and /usr/local/opt. Copy only non-system ones
+dep_list()
+{
+    for lib in $(otool -X -L "$1" | grep -o '/usr/local/[^ ]*'); do
+        found=0
+        for l in "${dep_libs[@]}"; do
+            if [ "$l" = "$lib" ]; then
+                found=1
+                break
+            fi
+        done
+        if [ $found -eq 0 ]; then
+            dep_libs+=( "$lib" )
+            dep_list "$lib"
+        fi
+    done
+}
+
+dep_list "$REPO/$NAME/cv2/"cv2*.so
+
+if [ ! -d "$REPO/$NAME/cv2/.dylibs" ]; then
+    mkdir -p "$REPO/$NAME/cv2/.dylibs"
+fi
+for lib in "${dep_libs[@]}"; do
+    cp -f "$lib" "$REPO/$NAME/cv2/.dylibs"
+done
+
+if [ $ENABLE_HEADLESS -eq 0 ]; then
+    dep_libs=()
+    dep_list /usr/local/Cellar/qt/5.15.2/plugins/platforms/libqcocoa.dylib
+    if [ ! -d "$REPO/$NAME/cv2/qt/plugins/platforms" ]; then
+        mkdir -p "$REPO/$NAME/cv2/qt/plugins/platforms"
+    fi
+    cp /usr/local/Cellar/qt/5.15.2/plugins/platforms/libqcocoa.dylib "$REPO/$NAME/cv2/qt/plugins/platforms"
+
+    for lib in "${dep_libs[@]}"; do
+        cp -f "$lib" "$REPO/$NAME/cv2/.dylibs"
+    done
+fi
+
+cp -f "$REPO/$NAME/opencv/data/haarcascades/"*.xml "$REPO/$NAME/cv2/data"
+# Copy licenses
+cp -f "$REPO/$NAME/LICENSE.txt" "$REPO/$NAME/LICENSE-3RD-PARTY.txt" "$REPO/$NAME/cv2"
+
+cd "$REPO/$NAME"
+# FIXME: Is there a way to package wheel without setup.py?
```

Comment:
Ok, now the next lines seem strange to me. As I understand, you have 2 options:
- build the project with CMake (that you've already done by this line), copy all the system dependencies at the proper places (done too), and zip the folders structure to get the archive (which should be named properly and has .whl extension)
- build the project with CMake, prepare the setup.py that knows nothing about building the project, but just specifies the dependencies and describes the already built libraries (to put them into package). Then, you run python3 setup.py bdist_wheel and get the wheel package, but without system dependencies. And the next step is to call `delocate-wheel` on OSX or `auditwheel repair` on Linux to package the system dependencies.